protocol/xmrtaker/instance.go

```diff
@@ -100,22 +100,36 @@ func (inst *Instance) createOngoingSwap(s swap.Info) error {
 			return err
 		}
 
-		inst.backend.XMRClient().Lock()
-		defer inst.backend.XMRClient().Unlock()
-
-		// TODO: do we want to transfer this back to the original account?
-		addr, err := monero.CreateWallet(
-			"xmrmaker-swap-wallet",
-			inst.backend.Env(),
-			inst.backend.XMRClient(),
-			kp,
+		conf := inst.backend.XMRClient().CreateABWalletConf("xmrtaker-swap-wallet-db-restored")
```

Comment:
maybe put everything in this `if err == nil` block into its own func? there's a lot going on

Reply:
This is the code that is generating the view key from the spend key. It needs a major rewrite and better unit test coverage in a follow-up PR. Let's break it up, but do it in the next PR.